src/FixedDataTable.js

```diff
@@ -932,25 +953,23 @@ class FixedDataTable extends React.Component {
       reorderColumn: columnKey,
     });
 
-    if (scrollStart !== scrollX && onHorizontalScroll) {
-      onHorizontalScroll(scrollX)
+    if (scrollStart !== scrollX) {
+      this._scrollToX(scrollX)
     };
   }
 
   _onScroll = (/*number*/ deltaX, /*number*/ deltaY) => {
     const {
       maxScrollX,
       maxScrollY,
-      onHorizontalScroll,
-      onVerticalScroll,
-      scrollActions,
       scrollFlags,
       scrollX,
       scrollY,
       scrolling,
     } = this.props;
     const { overflowX, overflowY } = scrollFlags;
 
+    this._didScrollStop.reset();
     if (!scrolling) {
       this._didScrollStart();
```

Comment:
Also I'm thinking if we should check if an actual scroll resulted from this and then call this._didScrollStart and this._didScrollEnd (the if's down below may not be true always)